parts/windows/windowscsehelper.ps1

```diff
@@ -183,7 +183,8 @@ function DownloadFileOverHttp {
     )
 
     # First check to see if a file with the same name is already cached on the VHD
-    $fileName = [IO.Path]::GetFileName($Url)
+    $cleanUrl = $Url.Split('?')[0]
```

Comment:
I'm not sure why we need this change. In the past, we may need to use url with SAS token which contains "?". It seems that we don't use url that contains "?" now so maybe this change will not influence anything.

Reply:
We use private storage account in tests. So, I need to use SAS keys to download the archive in tests.

Reply:
I tried to assign an MSI to the VM. But it seems like more changes are required in the windows CSE to use it. And I'm not confident enough to change it.

Reply:
Current windows e2e are using MSI with permission to upload and download files from a storage account. An example is here: https://github.com/Azure/AgentBaker/blob/master/e2e/windows/upload-cse-logs.ps1#L11

Reply:
Got it. Thank you for the clarification.